ansible_collections/arista/avd/plugins/action/validate_and_template.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible.errors import AnsibleActionFail
+from ansible.plugins.action import ActionBase
+from ansible.utils.display import Display
+
+from ansible_collections.arista.avd.plugins.filter.add_md_toc import add_md_toc
+from ansible_collections.arista.avd.plugins.plugin_utils.schema.avdschema import AristaAvdError, AvdSchema
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import compile_searchpath, template
+
+VALID_CONVERSION_MODES = ["disabled", "warning", "info", "debug"]
+VALID_VALIDATION_MODES = ["disabled", "error", "warning", "info", "debug"]
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        # Validate Arguments
```

Comment:
Maybe lets move the argument validation to a separate function as it is a lot of code for not much in the main run.

Any reason not to use the built-in argument spec? https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec

Reply:
Because it does not work in our oldest supported versions of Ansible.